src/object.c

```diff
@@ -398,16 +398,19 @@ void decrRefCount(robj *o) {
     }
 }
 
-/* See dismissObject() */
+/* See dismissObject(). sds is an exception. Instead of dismissing it
+ * with madvise(MADV_DONTNEED) we free it via the allocator. This has
+ * advantage that it allows the allocator to accumulate free buffers
+ * to free whole pages. While madvise is nop if the buffer is less
+ * than a page. The allocator overhead is minimal, because raw strings
+ * (sds) use the fast path. */
```

Comment:
The important part to mention, and why sds is an exception and why the allocator overhead is minimal, is that the size is known.

```suggestion
/* See dismissObject(). sds is an exception, because the allocation
 * size is known. Instead of dismissing it with madvise(MADV_DONTNEED)
 * we free it via the allocator, which has minimal overhead when the
 * size is known. This has advantage that it allows the allocator to
 * accumulate free buffers to free whole pages, while madvise is nop
 * if the buffer is less than a page.  */
```